extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBinding.java

```diff
@@ -67,14 +68,11 @@ public ServiceBinding(Path bindingDirectory) {
             return Collections.emptyMap();
         }
 
-        File[] files = directory.toFile().listFiles(new FileFilter() {
-            @Override
-            public boolean accept(File f) {
-                try {
-                    return !Files.isHidden(f.toPath()) && !Files.isDirectory(f.toPath());
-                } catch (IOException e) {
-                    throw new IllegalStateException("Unable to determine if file '" + f + "' is a regular file", e);
-                }
+        File[] files = directory.toFile().listFiles(f -> {
```

Comment:
This is intentional. We avoid lambdas in runtime code as they have a cost. Please revert this change.

Reply:
I've reverted this change.